js/ui/html_editor/utils/image_uploader_helper.js

```diff
@@ -0,0 +1,459 @@
+                const $widthEditor = $('<div>').appendTo($content);
+
+                this.widthEditor = this.createKeepAspectRatioEditor($widthEditor, data, 'height');
+
+                const $ratioEditor = $('<div>').appendTo($content);
+
+                this.module.editorInstance._createComponent($ratioEditor, ButtonGroup, {
+                    items: [{
+                        icon: 'link',
+                        value: 'keepRatio',
+                    }],
+                    hint: 'Keep aspect ratio',
+                    keyExpr: 'value',
+                    stylingMode: 'outlined',
+                    selectionMode: 'multiple',
+                    selectedItemKeys: ['keepRatio'],
+                    onSelectionChanged: (e) => {
+                        this.shouldKeepAspectRatio = !!e.component.option('selectedItems').length;
+                    }
+                });
+
+                return $content;
+            } },
+            { dataField: 'height', colSpan: 5, label: { text: localizationMessage.format(DIALOG_IMAGE_FIELD_HEIGHT) }, template: (data) => {
+                const $content = $('<div>');
+
+                this.heightEditor = this.createKeepAspectRatioEditor($content, data, 'width');
+
+                return $content;
+            } },
+            { dataField: 'alt', colSpan: 11, label: { text: localizationMessage.format(DIALOG_IMAGE_FIELD_ALT) } }
+        ];
+    }
+
+}
+
+class UpdateUrlStrategy extends AddUrlStrategy {
+    constructor(module, config, formData) {
+        super(module, config);
+        this.formData = formData;
+        this.modifyFormData();
+    }
+
+    modifyFormData() {
+        const { imageSrc } = this.module.quill.getFormat(this.selection.index - 1, 1);
+
+        if(!imageSrc || this.selection.index === 0) {
+            this.selection = {
+                index: this.selection.index + 1,
+                length: 0
+            };
+            this.module.quill.setSelection(this.selection.index, this.selection.length, SILENT_ACTION);
+        }
+    }
+
+    pasteImage(formData, event) {
+        this.module.quill.deleteText(this.embedFormatIndex(), 1, SILENT_ACTION);
+        this.selection.index -= 1;
+        super.pasteImage(formData, event);
+    }
+
+    embedFormatIndex() {
+        const selection = this.selection ?? this.module.quill.getSelection();
+
+        if(selection) {
+            if(selection.length) {
+                return selection.index;
+            } else {
+                return selection.index - 1;
+            }
+        } else {
+            return this.module.quill.getLength();
+        }
+    }
+}
+
+class BaseFileStrategy extends BaseStrategy {
+    closeDialogPopup(editorInstance, data) {
+        editorInstance._formDialog.hide({ file: data.value ? data.value[0] : data.file }, data.event);
+    }
+
+    serverUpload() {}
+
+    isBase64Editable() {
+        return false;
+    }
+
+    getUseBase64() {
```

Comment:
There is a lot of code related only to base64 strategy. Imho, it should be placed in an appropriate strategy, not here in base one